File: Verde/Exercícios/Pali.java
package Verde.Exercícios;

import java.util.*;

public class Pali {

    // função booleana para testar
    public static boolean isPali(String palavra) {
        // definir dados
        boolean is = true;
        int x = 0;
        int y = palavra.length() - 1;
        char a = 'a';
        char b = 'b';
        // metodo iterativo
        while (x <= y) {
            a = palavra.charAt(x);
            b = palavra.charAt(y);
            if (a != b) {
                is = false;
            }
            x++;
            y--;
        }
        // retornar
        return is;
    }

    // função main
    public static void main(String args[]){
        //definir dados
          Scanner scanner = new Scanner(System.in);
          String palavra;
          boolean is = false;
        //ler palavra e chamar função
          do{
             palavra = scanner.nextLine();
             if(!palavra.equals("FIM")){
                is = isPali(palavra);
                if(is){
                    System.out.println("SIM");
                }
                else{
                    System.out.println("NAO");
                }
             }
          }while(!palavra.equals("FIM"));
        //fechar scanner
          scanner.close();
          
      }

}

File: Verde/TP01/Q02/Pali.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <stdbool.h>
// Função Booleana que retorna se é Pali
bool isPali(char *palavra)
{
  // definir dados
  bool is = true;
  int tam = strlen(palavra);
  int x = tam - 2;
  int y = 0;
  // passar letras
  while (y <= x)
  {
    if (palavra[x] != palavra[y])
    {
      is = false;
    }
    x = x - 1;
    y = y + 1;
  }
  // retornar
  return (is);
}
// Função Main

int main()
{
  // definir dados
  char palavra[80];
  bool pali = false;
  fgets(palavra, sizeof(palavra), stdin);
  // chamar função enquanto diferente de FIM
  while (strcmp(palavra, "FIM\n") != 0)
  {

    pali = isPali(palavra);
    if (pali)
    {
      printf("SIM\n");
    }
    else
    {
      printf("NAO\n");
    }
    fgets(palavra, sizeof(palavra), stdin);
  }

  // retornar
  return (0);
}